### DISCUSSION TITLE
2024-01-19 - Explore GPTs (aka custom GPTs) - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by lukecookssw** [SSWConsulting/SSW.Rules.Content#7796] [Chewing the Fat]
Hey Proompters,

Let's talk about [this rule](https://www.ssw.com.au/rules/monetize-gpt-models/)

**Comment by camillars** [comment]
Custom GPTs are impressive. I am not a dev, but I can see it sticking around for sure 🙂

**Comment by matt-goldman** [comment]
The "SellMeThisPenPlugin" - I have mixed feelings about this. At first I was completely unimpressed. The title and copy it generated were cringe, and very obviously written by ChatGPT (at this stage I think we've moved beyond being able to actually use AI generated content; sure, be inspired by it but it's very easily identifiable now). But what impressed me was that it recognised the product I uploaded a picture of. Not just the product in a generic sense (it was a laptop), but the precise model, year, and configuration. That actually blew me away.

It then went on to search for pricing - this is also cool - but what would _really_ impress me here would be if it searched eBay and Marketplace listings and generated the ad content based on successful sales. E.g., items that sold quickest, or generated the most money within a given timeframe.

**Reply by Hona** [reply]
What about location, currency, what is actually successfully sold vs what is listed.
I feel the validation of its output will take longer than just doing it manually.

**Reply by sethdaily** [reply]
Agreed that would actually be impressive - I bet that's doable with ChatGPT's help connecting the APIs

**Comment by jakebayliss** [comment]
Dont have GPT 4 but this has made we want to get it!

**Comment by AlexRothwell** [comment]
I can see GPTs being quite useful in a lot of areas. In academic research, it could be used to summarise particular aspects of published papers as part of a literature survey. This could save the researcher a lot of time in the early stages and help identify where to focus more manual efforts

**Comment by leahy268** [comment]
I can see GPT's being very useful.  I could easily setup custom ones for myself.  LicensingGPT comes to mind for a start.  Negotiating the Microsoft Licensing stack is sometimes a very timeconsuming process

**Comment by SamBWagner** [comment]
Tried to do the task and it was dead. 

<img width="1241" alt="image" src="https://github.com/SSWConsulting/SSW.Rules.Content/assets/66365977/ce76cf46-905e-4470-8ca2-310109545d14">

**Figure: SellMeThisPenGpt | ChatGPT**

I don't think the store is long for this world. Might be cool but I think it'll likely follow the same trajectory as the Discord Store.

I like the idea of leveraging these models for features in larger products though.

**Comment by drwharris** [comment]
Cant see stores being valuable in this area. Anything that is worth real money will be monetised directly.

**Comment by MarcusIrmscherSSW** [comment]
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/129343825/f745969c-6f56-48f4-8574-0770814dd5e1)
I uploaded a stock image of a child and gave me a decent copy for the sale of striped t-shirt. I was impressed by its abilty to scan the competitors prices and its suggestions. However when i advised it "I want to sell the child not the t-shirt" it was less helpful.

**Reply by stephanfakossw** [reply]
It's not perfect yet... maybe in 4.5 😆

**Comment by joshbermanssw** [comment]
I do agree that tools that are valuable (and can be properly monotized) will be done so on a different site, however i do think this will be useful in a micro-action sort of way.

**Comment by Calinator444** [comment]
I hadn't really considered the idea that experienced developers could sell their prompts to others rather than the solutions they generate. Interesting.

**Comment by isaaclombardssw** [comment]
Never knew about the GPT store before now and it seems like a practical addition to the service, varied technologies being linked together as in the video is also interesting. The rule is fine, to help interpret the technology.

**Comment by tkapa** [comment]
This is a decently cool idea, but I'm not sure how it works in execution. 

My issue is that anyone who is good enough at prompting will just do it themselves, and it's difficult to come up with something valuable enough that someone couldn't do so. 

OpenAI also isn't complex enough to do the cool things. E.g. Recommend MTG cards based off of my decks, sourced from a Joint DB of cards from different sources.

**Comment by Hona** [comment]
I feel the main benefit is for sharing version controlled prompts among a team. (if it does that, I'm not sure)

e.g. the rule writer prompt we have, make it the app then any change the consumers get automatically.
Could be an improvement over copy pasting text that could get outdated

**Comment by YalalCakirSSW** [comment]
In my opinion, custom GPT brings a lot of opportunities to the current AI market. With the IoT systems, even maybe you can build your own Jarvis. With the GPT Market they can stay at the top of the AI market. But the change is rapid.

**Comment by bettybondoc** [comment]
It's an exciting thought to make money out of GPT models and also use products from the GPT store.

I'm quite impressed with 'SellMeThisPen,' especially its ability to identify the product and generate fairly accurate pricing. I intentionally took a blurred photo of a rubber ducky (similar to what you might find in a casual Facebook Marketplace post) and it didn't have any trouble identifying it and even gave me a range of price comparisons from sites like Etsy, eBay, and various bulk trading platforms.

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/33026270/e1876412-2c90-400e-b6c8-8dc175849353)
**Figure: A screenshot of an existing product photo**

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/33026270/ffe42a6d-76ea-441d-9f5a-a9461c3f2162)
**Figure: Trying to sell Ben's rubber ducky**

**Comment by sethdaily** [comment]
GPTs are handy for speeding up small actions.
Longevity - I think there will be much more value in custom solutions that use large language model APIs. Completely customizable and private.

**Comment by TobyChurches** [comment]
I like the idea of the GPT store, it provides a low barrier to entry for creating potentially useful GPT tools. Could be good to create one that will generate emails that conform to the SSW rules format 😅. 

Although the description of 'Sell Me This Pen' was clearly a ChatGPT output, it was interesting to see that it could provide a suggested pricing.

**Comment by jimmy1992abc** [comment]
For what AI can do - See some amazing tech news recently. Think about what we can do 3 years ago and what we can do now! The internet is changing at a ridiculous speed!

For GPT Store - I believe OpenAI is trying to improve its integration with other tech companies and generate some referrals. This is a short win and less effort. I believe there will be something more properly prepared in the future.

For the Website generated from the video - It looks like only the front end, how can people merge the website to their server and domain? Or you have to use a 3rd party like G*Da**y - then the only value GPT can bring to me is to promote a service that fits the job.

**Comment by jeoffreyfischer** [comment]
It's crazy seeing how fast these GPT AI technologies are growing!

**Comment by jaydenalchin** [comment]
I like the GPT store as a way to find new ways to use ChatGPT, but I don't see it being a long-lasting heavily used tool in this very quickly evolving landscape.

**Comment by amankumarrr** [comment]
I believe GPT is just the beginning of the AI revolution. Over time, companies will develop their own models, challenging existing GPT dominance.

**Comment by vladislav-kir** [comment]
GPT Store seems like an interesting idea, but I am skeptical about it. I quite doubt it would be a long term thing, and I think most of the really useful things would be directly monetized.

**Comment by apforsyth** [comment]
Cool!  but I doubt it would replace the app itself in the short term

**Comment by cvionnet** [comment]
That's a nice feature, but not 100% convinced. My first try using a custom GPT for board game rules wasn't great; it missed many rules in the game tested. Maybe we need a custom GPT to find the best custom GPT!

**Comment by AntPolkanov** [comment]
It seems like OpenAI will be moving towards customizations and/or privatization of GPTs.
Many companies want to have a GPT which knows the very details of their business and is able to assist customers.
I bet that there soon will be an option to plug-in your custom GPT on a website as a chatbot.

**Comment by piers-sinclair** [comment]
GPTs are pretty awesome, but I think they need to do a bit more to get it ready for enterprise use. The new Teams subscription might solve that, but I haven't had a chance to play with it yet!

**Comment by synnoveisabel** [comment]
It's a lot of fun exploring the GPT Stores, and see how much you can do with AI. It's going to be interesting seeing how far the GPTs will go in the future.

**Comment by ricksu978** [comment]
Did a test on selling the Mi Band.

✅ GPT created placeholder text for unsure fields, e.g. Brand, Country and Price
✅ GPT also search prices for all different brands.

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/35659961/91c2d92d-c838-4bb2-9ddb-32d581d7a76a)

# Stay Connected, Stay Fit!
Join the fitness revolution with this sleek, smart fitness tracker! Perfect for keeping up with your health goals this season.

# Specification
Category: Wearables/Electronics
Brand: [Placeholder for Brand]
Color: Black
Country of origin: [Placeholder for Country]
Material: Silicone (band)
Price suggestion
I will now search for the second-hand prices for similar smart fitness trackers to suggest a price. Please wait a moment.

From a [quick search](https://chat.openai.com/backend-api/bing/redirect?query=smart+fitness+tracker+used+price), here's what I found.

Based on the information gathered from the sources, here's a price suggestion for the smart fitness tracker:

Xiaomi Mi Smart Band 8: £39.99/€49.99​​
Apple Watch Series 9 (Used): $237.59 - $349​​​​
Samsung Galaxy Watch 6: $295​​
Fitbit Charge 6: ~$130 (discounted from $160)​​

**Comment by PennyWalker** [comment]
This is the first time I've been truly sad about still using 3.5, I really wanted to give it a go after experimenting with last weeks generator! I saw the one it created for Levi and it was good bones but needed a lot of work to make it ready to go. Great way of perhaps getting a simple template for someone not very technical.

**Comment by bahjatalaadelssw** [comment]
Most AI applications lack a future sustainability focus, it's still in a sandbox environment the major companies might not move to sweeping use cases.

**Comment by tombui99** [comment]
These custom GPTs are definitely very impressive and I can see their usefulness in many areas especially in the field of software development. This could save a lot of time when designing and building a new software product.

**Comment by JudyMao99** [comment]
It's so nice to know those rules, to be honest, I have just started using ChatGPT (ChatGPT 3.5). I am interested in trying GPT 4.0 now.🤩

**Comment by ncn-ssw** [comment]
A niche use for the custom GPTs could be creating images with a consistent style. Below is an example image generated from a custom GPT that @jernejk and I created for AI Hack Day Brisbane 2023.

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/132531112/0a5bc423-4314-4398-88c4-cf630607d7e0)
**Figure: Ship Builder Original GPT at time of creation**

**Comment by Ben0189** [comment]
Even though it doesn't perform perfectly for certain models (not at the point where I'm willing to pay and use it daily), but it's still impressive for now. Hopefully, they can keep making it better.

**Comment by JackDevAU** [comment]
GPT 4 still Cool ✅
OpenAI Store - Cool ✅

SellMeThisPen - pretty cool but I feel a bit useless for real world scenarios. If it showed a bit more sell data on similar products that would be amazing.

**Comment by tomek-i** [comment]
Because it is so easy t o make your own GPT it won't take long and the store will be saturated with crapware. Like all those scam crypto that are based on ETH (because Ethereum and smart contracts made it so easy to launch your own crypto)

The GPT store will need to be regulated like the iOS Apple Store not like the android store full of malware.

I'm still excited to see how everything will develop and move forward but I feel already the market gets over saturated.

https://dang.ai/ai-graveyard

**Comment by divgul** [comment]
Custom GPTs look really cool. I didn't have GPT-4 but this CTF makes me want to subscribe to it. 

SSW should definitely consider creating a custom GPT. The idea of a SoftwareConsultantGPT sounds very cool. What Knowledge base can we train it on? Spec Reviews perhaps?  

TechTeacherGPT - SSW could even build a GPT that teaches tech. SSW has great domain knowledge in many technologies which we could use to train this GPT. For e.g. our knowledge on Clean Architecture or Azure could be fed into it. We'll need to think about how to make it profitable though as an ambitious custom GPT would require a lot of time and effort to build.

**Comment by Levijj22** [comment]
I love creating GPT's! However I can't imagine paying for one as i find they are still temperamental.

**Comment by steven0x51** [comment]
Don't have GPT-4 yet, but this convinced me to try it

**Comment by jernejk** [comment]
I've built a couple of custom GPT plugins in the last few months and I mostly use them as shortcuts.

I've made:
- Extremely negative code review (force AI to believe your code until it disagrees with you)
- .NET Consultant
- EF Core specialist

And a couple of others.

I try to be as independent as possible from AI and I use it only to verify my work or generate fluff that nobody reads.

**Comment by CoffeeBugHub** [comment]
GPT is going through customization; I think it is cool to see that there is so much more you can do with GPT.

**Comment by stephanfakossw** [comment]
Quite impressive and cool

**Comment by AshAnil** [comment]
GPT-4 is incredibly impressive. I like the idea of consolidating all AI applications into a single explorer page.

**Comment by chrisschultzssw** [comment]
Whether the GPT Store has longevity or not, we're seeing bigger brands exploring ways to connect with users through AI (and of course monetise it).

**Comment by MattParkerDev** [comment]
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/61717342/096c24e8-f796-4bce-9239-a97fa662d4cd)
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/61717342/10bcdc72-771a-489a-aeff-7a3d3af00bf6)
Completely mis-gauges the age of the car.
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/61717342/d688a9db-94d4-412e-a157-2e41002d5779)
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/61717342/fccfd708-2699-410e-83f3-dbdca0ff8079)
Reprompting gives results for FJ60's (its actually a HJ75).
Doesn't feel useful, I'm prompt engineering a google search basically.

I'm also not sure that the linked SSW Rule needed to be a rule? 👍🏼 if you agree.

**Comment by JeanThirion** [comment]
That thing was bloody impressive !!

**Comment by christoment** [comment]
Very sad that they put this behind a paid GPT-4 but maybe this is one of the reason they launch this service - to drive more users in GPT-4 😅

**Comment by UlyssesMaclaren** [comment]
For my own productivity, I've been creating GPTs instead of having prompt templates... they're just more easily accessible. 

I wouldn't bother trying to make ones that are popular on the store... just make ones that are hyper-specific to what you need, and share them with your own company.

e.g. the best 2 atm are for post initial meetings, and for writing rules.

See all the ones we've made so far are here: https://sswcom.sharepoint.com/SitePages/SSW-GPTs.aspx

**Comment by NadeekaJK** [comment]
Keen to experiment with GPTs, I see a lot of potential.

**Comment by AttackOnMorty** [comment]
One store to rule them all.

**Comment by ozdemir-mehmet** [comment]
Using regular GPT/Copilot/Bard seems plenty to me.

**Comment by calumjs** [comment]
I think long term LLMs will replace the browser entirely, which will totally change everything. The idea of going to a custom gpt on a web page will seem silly and backwards. But for now it's cool to play with.

**Comment by Marxoz** [comment]
I like the idea of GPTs, but there is still a lot lacking for it to be completly useful and robust. Just being able to have a openAI userId passed to a paid will already provide a lot of value to developers. 

I did have different responses for the same input using a GPT I created. So this might also be an issue for some, where the output needs to be identical for different users in say, the same company. But these are small issues that have work arounds.

**Comment by ozair-ashfaque** [comment]
Building apps from GPT is awesome, but putting cost on it can be a deal breaker for some developers.

**Comment by miichaelsmedley** [comment]
I like GPT but find the text very verbose at times and a little more that is needed, and as such it does more than is required. I often have to ask it to be more succinct.

**Comment by jackreimers** [comment]
I think we are going to see a lot of junk custom GPTs in the coming months. GPT is awesome and very powerful but it is not a catch all solution and often people use it for things where it does not make sense.

In saying that I think opening up the training and sharing of custom models is a great move and once the dust settles there will be a lot of cool stuff to be found.

**Comment by william-liebenberg** [comment]
GPT Stores will be like the new App Stores... just wait for Microsoft Store or Apple Store to have a GPT section :)

**Comment by jerryluossw** [comment]
GPT Store is so cooooooooooool, unfortunately, you can't easily use it in China.
I think it will always be there until we have Artificial General Intelligence.

**Comment by hveraus** [comment]
It is good to see OpenAI is creating it's own eco-system, I hope we can see more and more useful applications in the store.

**Comment by sylhuang** [comment]
I'm curious to see what becomes the most popular needs in store as it matures. Definitely impressive.

**Comment by Aibono1225** [comment]
Love to see the ecosystem grow, though I get a feeling that we would have a more convenient tool to replace GPT Stores

**Comment by jimmidier** [comment]
I think custom GPTs only shine when they can be proactively fine tuned with unsupervised data from the user, instead of just pre prompting pre trained models and stay where they are forever. After all it's a lot to imagine if a GPT developer could say, here's the SellMeThisPen GPT I designed, go ahead and bring it with you on your daily work, and you'll get a marketing expert within a month.

**Comment by Berber-Mama** [comment]
I didn't expect the store in such a short time, this is impressive. I wish I could try them without limit in the future here.

**Comment by KristenHu** [comment]
The GPT Store is definitely impressive, but I don't think it would be considered epoch-making due to the rapid updates and advancements in AI products.

**Comment by tino-liu** [comment]
I don't currently have GPT-4, but man this is so impressive that I'm gonna go get one.

**Comment by GordonBeeming** [comment]
I received some advise if I wanted to sell my SSW Travel Mug on a marketplace, some interesting information was shared on how it got the results

**Comment by danielmackay** [comment]
I didn't know Custom GPT's were so easy to create.  I think I'd need to play around with building a few to better cement my understanding of their applicability.

I thought the "Sell My Pen" GPT was pretty cool!

**Comment by kenzengssw** [comment]
GPT Store is awesome, I've tried some extension apps from it, which was impressive.  I'm genuinely excited about the future of its ecosystem.

**Comment by PothieuG** [comment]
CustomGPT are so easy to create. What a step up !

**Comment by JackDuan123** [comment]
I have no interest on GPT4-GTPn,
The truth is harsh, OpenAI has no future, because nobody will want to contribute to it anymore, it's close source. the next AI giant will be either Meta or hugging face.

More and more evidences have shown that open-source models can perform as good.

We should market on hugging face because hugging face work very closely to Azure, and officaillty support deploy AI models from Hugging face repository to Azure, which is the pioneer of MLOps or DLops

Something to read or watch:

I have open source RAG solution that can be 100% self hosted or cloud hosted and it's free , why would I bother using GPT4:
https://github.com/run-llama/rags

here is a example of a case from youtube, very cool:
https://www.youtube.com/watch?v=SedGB8m2XLM

I think many comanies will eventually go to self-hosting for security reason, we should just simply stop doing marketing on ChatGPT, it is not good.

We should marketing on things like, deploy your customized or fine-tuned model to Azure:
https://huggingface.co/blog/hugging-face-endpoints-on-azure

Once we have RAG pipelines and train model automatically occasionally on that pipelines, deplpy it to cloud (Azure) using yaml, or MLOps, will be the next cool things

**Reply by PothieuG** [reply]
Very interesting Jack !
I like your response to the topic, and I'm convinced as well that the truth will be in open source solutions :)

Did you heard about [Miastral.ai](https://mistral.ai/) ?

I've got the feeling that this will be the next giant : 
- open source
- light weight
- can be use locally (on a good config though)
- the 7go model (that can be used with a 16Vram video card) gives sames responses (sometimes better) than GPT 3.5 turbo

Have a look !

**Comment by BrookJeynes** [comment]
I believe long term the GPT store will become just like NPM, some good things but a whole lot of junk

**Comment by babakamyljanovssw** [comment]
I don't have GPT-4 currently, but GPT store sounds promising. Excited to give a try for GPT store features!
